### PR TITLE
feat: [PL-31161]: Aggregate API for Streaming Destination Listing

### PR DESCRIPTION
**Pull request by reetika04**
## Describe your changes

Aggregate API for Streaming Destination Listing - Has information about connector.

Testing:

- Tested in local env.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- Immutable delegate `trigger publish-delegate`
</details>

<details>
  <summary>PR Check triggers</summary>

You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ti0, ti1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
    - CodeFormat: `trigger codeformat`
    - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/43193)

### CODE REVIEW

**Review by nishantsaini20 — CHANGES_REQUESTED** [review]
> Reviewed 3 of 6 files at r1, 2 of 5 files at r2, 4 of 5 files at r3, 1 of 1 files at r4, 1 of 1 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43193)* status: 10 file(s) to review. 1 change requests. 0 of 2 approvals obtained. 

___
*[`platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml` line 348 at r4](https://reviewable.io/reviews/harness/harness-core/43193#-NN2OBfd8ALxUKKTUjj3:-NN2OBfd8ALxUKKTUjj4:b-hfroy1) ([raw file](https://github.com/harness/harness-core/blob/a3d3e611b0535215131b2327183eff6901265574/platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml#L348)):*
> ```YAML
>       description: Streaming Destination Aggregate DTO
>       properties:
>         streamingDestination:
> ```

use snake case, streaming_destination
mark required
___
*[`platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml` line 350 at r4](https://reviewable.io/reviews/harness/harness-core/43193#-NN2OE6p7blwCZ8SgHW9:-NN2OE6p7blwCZ8SgHWA:bqfcol3) ([raw file](https://github.com/harness/harness-core/blob/a3d3e611b0535215131b2327183eff6901265574/platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml#L350)):*
> ```YAML
>         streamingDestination:
>           $ref: '#/components/schemas/StreamingDestinationDTO'
>         connectorInfo:
> ```

same, snake case
mark required
___
*[`platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml` line 352 at r4](https://reviewable.io/reviews/harness/harness-core/43193#-NN2OGRR5qifW2Eq2Vwd:-NN2OGRR5qifW2Eq2Vwe:b-m0bjf5) ([raw file](https://github.com/harness/harness-core/blob/a3d3e611b0535215131b2327183eff6901265574/platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml#L352)):*
> ```YAML
>         connectorInfo:
>           $ref: '#/components/schemas/Connector'
>         streamingDetails:
> ```

same, snake case
___
*[`platform-service/modules/audit-service/src/main/java/io/harness/audit/api/streaming/impl/StreamingServiceImpl.java` line 97 at r4](https://reviewable.io/reviews/harness/harness-core/43193#-NN2Oeuz6s6UV8ezozgT:-NN2Oeuz6s6UV8ezozgU:bq2mqxk) ([raw file](https://github.com/harness/harness-core/blob/a3d3e611b0535215131b2327183eff6901265574/platform-service/modules/audit-service/src/main/java/io/harness/audit/api/streaming/impl/StreamingServiceImpl.java#L97)):*
> ```Java
>         connectorResourceClient.get(connectorRef.getIdentifier(), accountIdentifier, null, null));
>     if (connectorDTO.isEmpty()) {
>       String message = String.format("Connector with identifier [%s] is invalid.", connectorRef.getIdentifier());
> ```

Make message more clearer. Should say that it doesn't exist OR not found.



_Code quote:_
```Java
is invalid.
```

**Review by reetika04 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43193)* status: 10 file(s) to review. 1 change requests. 3. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml` line 350 at r4](https://reviewable.io/reviews/harness/harness-core/43193#-NN2OE6p7blwCZ8SgHW9:-NN5T5EL-lOjM-tAUQjz:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/a3d3e611b0535215131b2327183eff6901265574/platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml#L350)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

same, snake case
mark required
</blockquote></details>

Done.
___
*[`platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml` line 352 at r4](https://reviewable.io/reviews/harness/harness-core/43193#-NN2OGRR5qifW2Eq2Vwd:-NN5T3fsEJI3xTt1ePxo:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/a3d3e611b0535215131b2327183eff6901265574/platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml#L352)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

same, snake case
</blockquote></details>

Done.
___
*[`platform-service/modules/audit-service/src/main/java/io/harness/audit/api/streaming/impl/StreamingServiceImpl.java` line 97 at r4](https://reviewable.io/reviews/harness/harness-core/43193#-NN2Oeuz6s6UV8ezozgT:-NN5T4Cy8roUK4tN7ID6:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/a3d3e611b0535215131b2327183eff6901265574/platform-service/modules/audit-service/src/main/java/io/harness/audit/api/streaming/impl/StreamingServiceImpl.java#L97)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

Make message more clearer. Should say that it doesn't exist OR not found.
</blockquote></details>

Done.

**Review by reetika04 — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43193)* status: 10 file(s) to review. 1 change requests. 4. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml` line 348 at r4](https://reviewable.io/reviews/harness/harness-core/43193#-NN2OBfd8ALxUKKTUjj3:-NN6AeQtD5lky7gAVc5k:b-896fix) ([raw file](https://github.com/harness/harness-core/blob/a3d3e611b0535215131b2327183eff6901265574/platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml#L348)):*
<details><summary><i>Previously, nishantsaini20 (Nishant Saini) wrote…</i></summary><blockquote>

use snake case, streaming_destination
mark required
</blockquote></details>

Done.

**Review by nishantsaini20 — DISMISSED** [review]
> Reviewed 1 of 2 files at r6, 1 of 1 files at r7, 2 of 2 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43193)* status: 10 file(s) to review. 1 of 2 approvals obtained.

File: platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml [hunk 1]
```diff
@@ -417,6 +510,15 @@ components:
...
             $ref: '#/components/schemas/StreamingDestinationCards'
+    StreamingDestinationAggregateListResponse:
+      description: Example response
+      content:
+        application/json:
+          schema:
+            type: array
+            items:
+              $ref: '#/components/schemas/StreamingDestinationAggregateDTO'
+      headers: {}
```
Comment (mankrit-singh) [inline platform-service/modules/audit-service/contracts/openapi/v1/openapi.yaml:521]: please add Pagination header

File: platform-service/modules/audit-service/src/main/java/io/harness/audit/api/streaming/impl/AggregateStreamingServiceImpl.java [hunk 2]
```diff
@@ -81,4 +107,56 @@ private long getFailedBatchCount(String accountIdentifier) {
...
+          if (lastStreamedBatch.isPresent()) {
+            StreamingBatchDTO streamedBatch = lastStreamedBatch.get();
+            BatchFailureInfo batchFailureInfo = streamedBatch.getFailureInfo();
+            streamingDestinationAggregateDTO.setStreamingDetails(
+                new StreamingDetails()
+                    .lastStreamedAt(streamedBatch.getLastStreamedAt())
+                    .errorMessage(batchFailureInfo == null ? "" : batchFailureInfo.getMessage())
+                    .status(streamingDestinationsApiUtils.getStatusEnum(streamedBatch.getStatus())));
+          }
+          return streamingDestinationAggregateDTO;
```
Comment (mankrit-singh) [inline platform-service/modules/audit-service/src/main/java/io/harness/audit/api/streaming/impl/AggregateStreamingServiceImpl.java:158]: please use ApiUtils.addLinkstoHeaders to populate Pagination header

**Review by mankrit-singh — COMMENTED** [review]
> Reviewed 3 of 6 files at r1, 1 of 5 files at r2, 2 of 5 files at r3, 1 of 2 files at r8, 3 of 3 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43193)* status: 3 file(s) to review. 0 of 2 approvals obtained.

**Review by mankrit-singh — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43193)* status: 3 file(s) to review. 1 of 2 approvals obtained.

**Review by nishantsaini20 — APPROVED** [review]
> Reviewed 3 of 3 files at r9, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/43193)* status: :shipit: complete! ✓

**Pull request merged**